v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlex.java

```diff
@@ -248,27 +252,74 @@ public static PipelineResult runAvroPipeline(
 
     WriteResult writeResult = null;
 
-    if (options.getAvroFormat().equals("NON_WIRE_FORMAT") && options.getAvroSchemaPath() != null) {
-
-      writeResult = kafkaRecords.apply(AvroTransform.of(options));
+    if (options.getMessageFormat().equals("AVRO_BINARY_ENCODING")
```

Comment:
nit: The if clauses are very long and hard to read/understand the logic. Consider splitting this method into others even further, e.g.
```
  if (options.getMessageFormat().equals("AVRO_BINARY_ENCODING") {
    runAvroBinaryEncodingPipeline(...);    
  } else if (options.getMessageFormat().equals("AVRO_CONFLUENT_WIRE_FORMAT") {
    runAvroConfluentEncodingPipeline(...);    
  } else {
    throw UnsupportedOperationException("Message format " + options.getMessageFormat() + " is unspported.");
  }
```

I also suggest following the above "if/else-if/else throw exception" pattern instead of just if/else because if the latter may hide problems in the future when a 3rd format is added (which is very easy to forget to add here).

Reply:
Also, what if (options.getMessageFormat().equals("AVRO_BINARY_ENCODING") && options.getBinaryAvroSchemaPath() == null)? Nothing happens?

Reply:
I will change it to if/else-if/else pattern. 

> Also, what if (options.getMessageFormat().equals("AVRO_BINARY_ENCODING") && options.getBinaryAvroSchemaPath() == null)? Nothing happens?

That is handled above, in this case it will never reach here.